MAINTAINING.md

```diff
@@ -0,0 +1,184 @@
+
+* Workflow files (`.github/workflows/*`). They are copied using rsync.
+* Dependabot configuration (`.github/dependabot.yml`).
+* Gradle wrapper.
+* `.gitignore`.
+* `ISSUE_TEMPLATE.md`, `LICENSE`, `MAINTAINING.md`, `config/HEADER` and `config/spotless.license.java`.
+* Checkstyle's `config/checkstyle/checkstyle.xml` and `config/checkstyle/suppressions.xml`.
+
+Regarding the Gradle wrapper, the template repo checks every week if there is a new Gradle version. If there is, it will
+upgrade the wrapper in the template repo itself, and via the files sync workflow this gets propagated to all repos. This
+way we make sure we stay up-to-date regarding Gradle versions in all repos.
+
+## Releases
+
+The release process is highly automated and normally involves just publishing a GitHub release. But before you get there,
+there are some parts you need to understand first.
+
+First of all, all the repos have an automatic changelog generation mechanism: when a change is made to the repo 
+(a push event), it creates (or updates if there is already one) a draft release, calculating the next patch version. The
+release notes will contain pull requests merged and issues closed since the last release.
+
+When the module is ready for a new release, check the generated release notes, and make changes if needed (for example,
+you can add an introduction paragraph highligting some items included in the release). If the version you are going to
+publish is not a new patch version, but a new minor or major, update the release notes text to reflect the new version.
+If you are publishing a milestone or release candidate, check the pre-release checkbox.
+
+Note that the release tags must be preceded with `v`, e.g.: `v1.2.3`.
+
+Once you publish the GitHub release, the 
+[Release GitHub Action workflow](https://github.com/micronaut-projects/micronaut-project-template/blob/master/.github/workflows/release.yml)
+will kick off, performing the following steps:
+
+* Pre-release: sets the `projectVersion` property in `gradle.properties` to the release version, and commit and pushes 
+  the result.
+* Uploads and publishes to Bintray.
+* Generates documentation guide and publishes it to the `gh-pages` branch.
+* Sends a pull request to Core to update the BOM.
+* Post-release: 
+  * Determines the next patch version, and sets it as a `BUILD-SNAPSHOT` version.
+  * Closes the milestone that matches the release version, and creates a new one for the next patch.
+
+Note that there are repos (like Core, that has a lot of modules) where uploading and publishing in the same Gradle execution 
+fails. If you want to upload, but not publish, define a repo secret `BINTRAY_PUBLISH` and set its value to `false`. Then, 
+once the release workflow has finished, you need to manually trigger the Bintray Publish workflow through the GitHub UI.
+
+If everything goes well, you now need to manually trigger the Maven Central publishing workflow via the GitHub UI.
```

Comment:
Maybe add something like: "If there is an issue with the release it's important not to trigger the Maven Central publishing workflow because once we publish a version to Maven Central we can not change or remove it anymore. On the other hand, if something failed during the Bintray upload, docs publication,... we can manually log in into Bintray UI, delete the artifacts, and then restart the workflow."